lxd/device/unix_hotplug.go

```diff
@@ -203,9 +209,9 @@ func (d *unixHotplug) postStop() error {
 	return nil
 }
 
-// loadUnixDevice scans the host machine for unix devices with matching product/vendor ids
-// and returns the first matching device with the subsystem type char or block.
-func (d *unixHotplug) loadUnixDevice() *udev.Device {
+// loadUnixDevices scans the host machine for unix devices with matching product/vendor ids
+// and returns the matching devices with subsystem types of char or block.
+func (d *unixHotplug) loadUnixDevices() []*udev.Device {
```

Comment:
can we return just `[]udev.Device` ?

Reply:
Yes, we can just just dereference `device` when we append it to `matchingDevices`.